bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/discovery/FreeboxOsDiscoveryService.java

```diff
@@ -0,0 +1,279 @@
+
+    private @NonNullByDefault({}) ApiBridgeHandler bridgeHandler;
+    private @Nullable ScheduledFuture<?> backgroundFuture;
+    private @NonNullByDefault({}) ThingUID bridgeUID;
+    private @NonNullByDefault({}) ApiConfiguration configuration;
+
+    public FreeboxOsDiscoveryService() {
+        super(THINGS_TYPES_UIDS, DISCOVERY_TIME_SECONDS);
+    }
+
+    @Override
+    public void deactivate() {
+        super.deactivate();
+    }
+
+    @Override
+    public void setThingHandler(@Nullable ThingHandler handler) {
+        if (handler instanceof ApiBridgeHandler) {
+            bridgeHandler = (ApiBridgeHandler) handler;
+            bridgeUID = bridgeHandler.getThing().getUID();
+            configuration = bridgeHandler.getConfiguration();
+            activate(null);
+        }
+    }
+
+    @Override
+    public @Nullable ThingHandler getThingHandler() {
+        return bridgeHandler;
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        stopBackgroundDiscovery();
+        backgroundFuture = scheduler.scheduleWithFixedDelay(this::startScan, BACKGROUND_SCAN_REFRESH_MINUTES,
+                BACKGROUND_SCAN_REFRESH_MINUTES, TimeUnit.MINUTES);
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        ScheduledFuture<?> future = this.backgroundFuture;
+        if (future != null) {
+            future.cancel(true);
+            this.backgroundFuture = null;
+        }
+    }
+
+    @Override
+    protected void startScan() {
+        logger.debug("Starting Freebox discovery scan");
+        if (bridgeHandler.getThing().getStatus() == ThingStatus.ONLINE) {
+            try {
+                discoverServer();
+                discoverPhone();
+                Map<String, @Nullable LanHost> lanHosts = bridgeHandler.getApi().getLanManager().getHostsMap();
+                discoverRepeater(lanHosts);
+                discoverPlayer(lanHosts);
+                discoverVM(lanHosts);
+                if (configuration.discoverNetDevice) {
+                    discoverHosts(lanHosts);
+                }
+            } catch (FreeboxException e) {
+                logger.warn("Error while requesting data for things discovery : {}", e.getMessage());
+            }
+        }
+    }
+
+    private void discoverServer() throws FreeboxException {
+        final String FBX_DELTA_GW = "fbxgw7";
+
+        SystemConf config = bridgeHandler.getApi().getSystemManager().getConfig();
+
+        ThingUID thingUID = new ThingUID(
+                config.getBoardName().contains(FBX_DELTA_GW) ? THING_TYPE_DELTA : THING_TYPE_REVOLUTION, bridgeUID,
+                config.getSerial());
+        logger.debug("Adding new Freebox Server {} to inbox", thingUID);
+        Map<String, Object> properties = new HashMap<>();
+        properties.put(Thing.PROPERTY_SERIAL_NUMBER, config.getSerial());
+        properties.put(Thing.PROPERTY_HARDWARE_VERSION, config.getBoardName());
```

Comment:
Please rather use config.getPrettyName() which corresponds to the hardware version displayed by the freebox OS WEB app.